packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp

```diff
@@ -109,14 +109,51 @@ namespace MueLu {
       const RCP<const FactoryBase>& myFactory = subFactories_[i];
       myFactory->CallBuild(coarseLevel);
       subBlockCoords[i] = coarseLevel.Get<RCP<dMV> >("Coordinates", myFactory.get());
-      //      subBlockMaps[i]   = subBlockCoords[i]->getMap();
+      subBlockMaps[i]   = subBlockCoords[i]->getMap();
     }
 
     // Blocked Map
-    RCP<const BlockedMap> coarseMap = Get< RCP<const BlockedMap> >(coarseLevel, "CoarseMap");
+    RCP<const BlockedMap> coarseCoordMapBlocked;
+
+    {
+      // coarseMap is being used to set up the domain map of tentative P, and therefore, the row map of Ac
+      // Therefore, if we amalgamate coarseMap, logical nodes in the coordinates vector would correspond to
+      // logical blocks in the matrix
+      RCP<const BlockedMap> coarseMap = Get< RCP<const BlockedMap> >(coarseLevel, "CoarseMap");
+      bool thyraMode = coarseMap->getThyraMode();
+
+      ArrayView<const GO> elementAList = coarseMap->getFullMap()->getNodeElementList();
+
+      LO blkSize = 0;
+      LO otherBlkSize;
+      for(size_t i=0; i<numSubFactories; i++) {
+        if (rcp_dynamic_cast<const StridedMap>(coarseMap->getMap(i, thyraMode)) != Teuchos::null){
+          otherBlkSize = rcp_dynamic_cast<const StridedMap>(coarseMap->getMap(i, thyraMode))->getFixedBlockSize();
+        } else {
+          otherBlkSize = 1;
+        }
+        if( i>0 && blkSize/i != otherBlkSize)
+          GetOStream(Warnings1) << "BlockedCoordinatesTransferFactory: Subblocks have different stride. Might cause problems..." << std::endl;
```

Comment:
Should this be an error instead of a warning?

Reply:
Was originally thinking of allowing block sizes to be different. I have changed it to be an error to avoid unexpected behavior.